test/e2e/specs/all-features-confirmation/errors/errors.spec.js

```diff
@@ -493,118 +445,170 @@ describe('Error related test cases,', function() {
+                  return modalActionBody.getText();
+              }).then(function (errorText) {
+                  expect(errorText).toBe(testParams.facetErrorstext.invalidPageCriteriaBody, "Error action text did not match");
+                  done();
+              }).catch(function(error) {
+                  console.log(error);
+                  done.fail();
+              });
+          });
+
+          it('On click of OK button the page should reload the page without paging conditions', function(done){
+              chaisePage.recordPage.getErrorModalOkButton().then(function(btn){
+                  browser.executeScript("arguments[0].click();",btn);
+                  return browser.driver.getCurrentUrl();
+              }).then (function(currentUrl) {
+                 recordsetPage = pageTestUrl.slice(0, pageTestUrl.search('@'));
+                 expect(currentUrl).toBe(recordsetPage, "The redirection to Recordset page failed");
+                 done();
+              }).catch( function(err) {
+                  console.log(err);
+                  done.fail();
+              });
+          });
 
-        it('After clicking back button initial page should appear', function(done){
-            chaisePage.recordPage.getErrorModalOkButton().then(function(btn){
-                return btn.click();
-            }).then (function (){
-                return browser.navigate().back();
-            }).then(function (){
-                return browser.driver.getCurrentUrl();
-            }).then (function(currentUrl) {
-                expect(currentUrl).toContain('id=269111', "The back button failed to go back to previous page.");
-                done();
-            }).catch(function(error) {
-                done.fail(error);
-            });
         });
 
-    });
+        describe("For no record found in RecordEdit app", function() {
 
-    describe("Dismissible Error Modal when using Navbar Delete button", function(){
+            beforeAll(function() {
+              browser.ignoreSynchronization = true;
+                url = browser.params.url + "/recordedit/#" + browser.params.catalogId + "/" + testParams.schemaName + ":" + testParams.table_name +  "/id=11223312121";
+                browser.refresh();
+                browser.get(url);
+            });
 
-        beforeAll(function() {
-            var url = browser.params.url + "/record/#" + browser.params.catalogId + "/" + testParams.schemaName + ":" + testParams.table_name +  "/id=2002";
-            browser.get(url);
-            deleteBtn  = chaisePage.recordPage.getDeleteRecordButton();
-        });
+            it('An error modal window should appear with Record Not Found title', function(done){
+              chaisePage.waitForElement(element(by.css('.modal-dialog '))).then(function() {
+                chaisePage.recordPage.getErrorModalTitle().then(function(modalTitle) {
```

Comment:
Let's use promise chaining here as well and add a catch handler with `done.fail()`:
```
  return chaisePage.recordPage.getErrorModalTitle()
}).then(function(modalTitle) {
  expect(modalTitle).toBe("Record Not Found", "The title of no record error pop is not correct");
  done();
}).catch(function(err) {
  console.log(err);
  done.fail();
});
```